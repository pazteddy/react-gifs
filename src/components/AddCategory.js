import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategorias }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e)=>{
    // console.log(e.target.value);
    setInputValue(e.target.value);
  }
  const onHandleSubmit = (e) =>{
    // Evita que se recargue la pagina
      e.preventDefault();
      // console.log('onHandleSubmit ',inputValue);
      if(inputValue.trim().length > 2){
        setCategorias(cats => [inputValue,...cats]);
        setInputValue('');
      }
      
  };
  return (
    <form onSubmit={ onHandleSubmit }>
      <input 
        type="text" 
        name="" 
        id=""
        value={inputValue}
        onChange={ handleInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategorias: PropTypes.func.isRequired
}
