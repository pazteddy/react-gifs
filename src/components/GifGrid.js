import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
 import { GifGridItem } from './GifGridItem';

export const GifGrid = ( {categoria}) => {
  // const [images, setImages] = useState([]);
  // //Solo se llame una vez para traer los gifs asi se actulice el componente
  // useEffect(() => {
  //   getGifs(categoria)
  //   .then( imgs => setImages( imgs));
  // }, [ categoria ]);

  //getGifs();

  const { data:images,loading } = useFetchGifs(categoria);
  
  return (
    <>
    <h3 className="animate__animated animate__fadeIn">{categoria}</h3>
    { loading && <p className="animate__animated animate__flash">Loading</p> }
    <div className="card-grid">
        {
          images.map( img => (
            <GifGridItem 
            key = {img.id}
            {...img}/>
          ))
        }
    </div>
    </>
  )
}
