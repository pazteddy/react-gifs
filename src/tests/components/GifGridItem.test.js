import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
describe(" Pruebas de <GifGridItem/>", () =>{
  const title = 'Un t√≠tulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url}/>);
  test('Debe mostrar correctamente <GifGridItem/>', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('debe tener un parrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Debe tener la imagen igual al url y el alt de los props', () => {
    const img = wrapper.find('img');
    // Para obtener una sola propiedad
    //console.log( img.prop('src'));
    // Obtener todas las propiedas
    //console.log( img.props());
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });
  
  test('debe tener animate__fadeIn', () => {
    const div = wrapper.find('div');
    //animate__fadeIn
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
  
  
  

});