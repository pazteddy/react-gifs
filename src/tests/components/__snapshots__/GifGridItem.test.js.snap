// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas de <GifGridItem/> Debe mostrar correctamente <GifGridItem/> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un título
  </p>
</div>
`;
